runtime/lua/vim/lsp/util.lua

```diff
@@ -1361,6 +1362,45 @@ local position_sort = sort_by_key(function(v)
   return {v.start.line, v.start.character}
 end)
 
+-- Gets the zero-indexed line from the given uri.
+-- For non-file uris, we load the buffer and get the line
+-- If a loaded buffer exists, then that is used.
+-- Otherwise we get the line using libuv which is a lot faster than loading the buffer.
+--@param uri string uri of the resource to get the line from
+--@param row number zero-indexed line number
+--@return string the line at row in filename
+function M.get_line(uri, row)
+  -- load the buffer if this is not a file uri
+  -- NOTE: when does this happen?
+  if uri:sub(1, 4) ~= "file" then
```

Comment:
I added this just to be sure, but when can this actually happen?

Relevant: #12357

Reply:
Some language servers like eclipse.jdt.ls or omnisharp provide the option to disassemble code or load the source code for dependencies on the fly. This is not part of the core specification and isn't enabled by default, but a plugin can opt-in to this feature. Once enabled, the language server will include entries with URIs like `jdt://...` in various results and the plugin needs a `BufReadCmd` autocmd to load those.

Reply:
In case you're curious:


https://user-images.githubusercontent.com/38700/119225851-9477a900-bb06-11eb-8b86-cce36cc25fac.mp4


(Tested it with this PR, so I can confirm that it still works)

Reply:
Ah awesome :) Thanks